Highway.cpp

```diff
@@ -13,22 +14,55 @@ void Highway::changeSpeed(int newSpeed)
 
 void Highway::addVehicleInternal(Vehicle* v)
 {
-    assert(false);
-
     /*
     depending on the derived type, call the member function that doesn't evade the cops. 
     */
+    /*
+    3) implement the Highway::addVehicleInternal 
+        this function should call the non-evasive member function of the derived class, so use the technique shown in the Casting video.
+    */
+    if( auto* car = dynamic_cast<Car*>(v))
+    {
+        car->closeWindows();
+    }
+    else if( auto* motorcycle = dynamic_cast<Motorcycle*>(v) )
+    {
+        motorcycle->lanesplitAndRace( 180 );
+    }
+    else if( auto* semitruck = dynamic_cast<SemiTruck*>(v) )
+    {
+        semitruck->pullOver();
+    }
```

Comment:
Really, you want the semi-truck to pull over when it enters the highway?
- [ ] revise.

Reply:
Added SemiTruck::cruiseInRightLane()